test/selectors.test.js

```diff
@@ -0,0 +1,91 @@
+import { createStore } from "redux"
+import { createBrowserHistory } from 'history'
+import { connectRouter, getLocation, createMatchSelector, getAction } from '../src'
+import { onLocationChanged } from '../src/actions'
+
+const push = pathname => onLocationChanged(
+  {
+    pathname,
+    search: '',
+    hash: '',
+  },
+  'PUSH'
+)
+
+describe("selectors", () => {
+  let store
+
+  beforeEach(() => {
+    const history = createBrowserHistory() 
+    const reducer = connectRouter(history)(() => {})
+    store = createStore(reducer)
+  })
+
+  describe("getLocation", () => {
+    it("gets the location from the state", () => {
+      const location = { pathname: "/", hash: '', search: '' }
+      store.dispatch(push('/'))
+      const state = store.getState()
+      expect(getLocation(state)).toEqual(location)
+    })
+  })
+
+  describe("getLocation", () => {
+    it("gets the location from the state", () => {
+      const action = "PUSH"
+      store.dispatch(push('/'))
+      const state = store.getState()
+      expect(getAction(state)).toBe(action)
+    })
+  })
+
+  describe("createMatchSelector", () => {
+    it("matches correctly if the router is initialized", () => {
+      const matchSelector = createMatchSelector("/")
+      store.dispatch(push('/test'))
+      const state = store.getState()
+      expect(matchSelector(state)).toEqual({
+        isExact: false,
+        params: {},
+        path: "/",
+        url: "/"
+      })
+    })
+
+    it("does not throw error if router has not yet initialized", () => {
+      const matchSelector = createMatchSelector("/")
+      const state = store.getState()
+      expect(() => matchSelector(state)).not.toThrow()
+    })
+
+    it("does not update if the match is the same", () => {
```

Comment:
For all test cases, instead of comparing match1 with match2, I think it will be easier to understand if we compare each one against its expected value.